crates/model/src/instruments/mod.rs

```diff
@@ -151,31 +278,88 @@ pub trait Instrument: 'static + Send {
         price: Price,
         use_quote_for_inverse: Option<bool>,
     ) -> Money {
-        let use_quote_for_inverse = use_quote_for_inverse.unwrap_or(false);
-        let (amount, currency) = if self.is_inverse() {
-            if use_quote_for_inverse {
-                (quantity.as_f64(), self.quote_currency())
+        let use_quote = use_quote_for_inverse.unwrap_or(false);
+        if self.is_inverse() {
+            if use_quote {
+                Money::new(quantity.as_f64(), self.quote_currency())
             } else {
-                let amount =
-                    quantity.as_f64() * self.multiplier().as_f64() * (1.0 / price.as_f64());
-                let currency = self
+                let amt = quantity.as_f64() * self.multiplier().as_f64() * (1.0 / price.as_f64());
```

Comment:
I think `amt` and `ccy` as local variable names are a little terse in this context and a bit different to other parts of the codebase. Lets stick with `amount` and `currency` for now.